src/main/java/eu/dissco/core/datacitepublisher/service/DataCitePublisherService.java

```diff
@@ -0,0 +1,311 @@
+        digitalSpecimen.getPid(),
+        digitalSpecimen.getReferentName(),
+        TYPE_DS,
+        digitalSpecimen,
+        getDescription(digitalSpecimen),
+        getSubjects(digitalSpecimen));
+  }
+
+  private DcRequest buildDcRequest(MediaObject mediaObject) {
+    return buildDcRequest(
+        mediaObject.get10320Loc(),
+        LANDING_PAGE_MO,
+        ALT_ID_TYPE_MO,
+        mediaObject.getPrimaryMediaId(),
+        mediaObject.getMediaHostName(),
+        mediaObject.getMediaHost(),
+        mediaObject.getIssuedForAgentName(),
+        mediaObject.getIssuedForAgent(),
+        mediaObject.getPidRecordIssueDate(),
+        mediaObject.getPid(),
+        mediaObject.getReferentName(),
+        TYPE_MO,
+        mediaObject,
+        getDescription(mediaObject),
+        getSubjects(mediaObject));
+  }
+
+  private DcRequest buildDcRequest(String xmlLoc, String landingPage, String altIdType,
+      String localId, String hostName,
+      String hostId, String issuedForAgentName, String issuedForAgentId, String pidRecordIssueDate,
+      String pid, String referentName, String dcType, Object fdoProfile,
+      List<DcDescription> descriptions, List<DcSubject> subjects) {
+    try {
+      var xmlLocs = xmlLocReader.getLocationsFromXml(xmlLoc);
+      var url = XmlLocReader.getLandingPageLocation(xmlLocs, landingPage);
+      var issueDate = getDate(pidRecordIssueDate);
+      return new DcRequest()
+          .withDcData(
+              new DcData()
+                  .withDcAttributes(new DcAttributes()
+                      .withAlternateIdentifiers(
+                          getAltIds(altIdType, localId))
+                      .withContributors(getContributors(hostName, hostId))
+                      .withCreators(getCreator(issuedForAgentName, issuedForAgentId))
+                      .withDates(getDates(issueDate))
+                      .withDescription(descriptions)
+                      .withDoi(getDoi(pid))
+                      .withPublicationYear(
+                          getPublicationYear(issueDate))
+                      .withRelatedIdentifiers(getRelatedIdentifiers(xmlLocs, url))
+                      .withSubjects(subjects)
+                      .withSuffix(getSuffix(pid))
+                      .withTitles(setTitles(referentName))
+                      .withType(getDcType(dcType))
+                      .withUrl(url))
+          );
+    } catch (InvalidFdoProfileRecievedException e) {
+      publishDlq(fdoProfile, pid);
+      throw new DataCiteMappingException();
+    }
+  }
+
+  private void publishDlq(Object message, String pid) {
+    String parsedMessage;
+    try {
+      parsedMessage = mapper.writeValueAsString(message);
+    } catch (JsonProcessingException e1) {
+      log.error(
+          "An error has occurred mapping given FDO profile to Datacite mapping, can not DLQ for fdo profile {}",
+          pid, e1);
+      kafkaPublisherService.sendDlq(pid);
```

Comment:
This is tricky as we would get a queue with both full messages as well as only PIDs. The idea of the DLQ is that it is the full message and we can retry the message. In this case I would just log it and it needs some manual action

Reply:
removed this dlq function altogether